src/use-swr-suspense.ts

```diff
@@ -0,0 +1,46 @@
+// TODO: add documentation
+
+import useSWR from './use-swr'
+
+const suspenseGroup = {
+  promises: [],
+  started: false
+}
+
+function _internal_useSWRSuspenseStart() {
+  if (suspenseGroup.started) {
+    suspenseGroup.started = false
+    throw new Error('Wrong order of SWR suspense guards.')
+  }
+  suspenseGroup.started = true
+  suspenseGroup.promises = []
+}
+
+function _internal_useSWRSuspenseEnd() {
+  if (!suspenseGroup.started) {
+    throw new Error('Wrong order of SWR suspense guards.')
+  }
+  if (!suspenseGroup.promises.length) {
+    suspenseGroup.started = false
+    return
+  }
+  suspenseGroup.started = false
+  throw Promise.race(suspenseGroup.promises).then(() => {
+    // need to clean up the group
+    suspenseGroup.promises = []
+  })
+}
+
+function useSWRSuspense(callback) {
+  _internal_useSWRSuspenseStart()
+  const data = callback(useSWR)
```

Comment:
In the PR you wrote this:
```js
const [user, teams] = useSWRSuspense(()=> {
  const { data: user } = useSWR('/api/user')
  const { data: movies } = useSWR('/api/movies')
  return [user, teams]
})
```
But from this code I understand it should be something like:
```js
const [user, teams] = useSWRSuspense(useSWR=> {
  const { data: user } = useSWR('/api/user')
  const { data: movies } = useSWR('/api/movies')
  return [user, teams]
})
```
Right? Does that mean if I'm importing `useSWRSuspense` I shouldn't import `useSWR` too? Could this cause an error if I'm using the imported version directly instead of the one `useSWRSuspense` gives me? From what I read there is nothing special about this useSWR call..

Reply:
Nope, both will work because in `callback(useSWR)` that's the exact same `useSWR` function.

Reply:
I just read the test below I understood why you need to call the `useSWR` you receive from `useSWRSuspense`.